src/plugins/intel_cpu/src/nodes/memory.hpp

```diff
@@ -20,20 +21,24 @@ namespace intel_cpu {
 namespace node {
 
 class MemoryOutput;
-class MemoryInput;
+class MemoryInputBase;
 
-class MemoryNode { //TODO , segregate interfaces
-    std::string _id;
+class MemoryNode {
  public:
-    explicit MemoryNode(std::string id) : _id(id) {}
+    explicit MemoryNode(std::string id) : m_id(id) {}
     explicit MemoryNode(const std::shared_ptr<ov::Node>& op);
     virtual ~MemoryNode() = default;
     std::string getId() const {
-        return _id;
+        return m_id;
     }
-    virtual void registerInputNode(MemoryInput*) = 0;
-    virtual void registerOutputNode(MemoryOutput*) = 0;
-    virtual void deregisterSibling(MemoryNode*) = 0;
+
+private:
+    std::string m_id;
+};
+
+class MemoryStateNode : public MemoryNode {
```

Comment:
Sounds MemoryStateNode can be merged into MemoryInputBase.

![image](https://github.com/openvinotoolkit/openvino/assets/24786748/e0cff8da-db01-4f9f-9f4c-8bfe28c8ff13)